mldsa/mldsa_native.h

```diff
@@ -448,9 +448,11 @@ typedef enum
  **************************************************/
 MLD_API_MUST_CHECK_RETURN_VALUE
 int MLD_API_NAMESPACE(signature_pre_hash_internal)(
-    uint8_t *sig, size_t *siglen, const uint8_t *ph, size_t phlen,
-    const uint8_t *ctx, size_t ctxlen, const uint8_t rnd[MLDSA_RNDBYTES],
-    const uint8_t *sk, mld_hash_alg_t hashAlg);
+    uint8_t sig[MLDSA_BYTES(MLD_CONFIG_API_PARAMETER_SET)], size_t *siglen,
```

Comment:
If there is an explicit length parameter we should stick to pointer notation instead of array notation. 
Please change that back to a pointer. I see that this is also wrong in other places - please fix them too.